contracts/clients/src/BlsWalletWrapper.ts

```diff
@@ -221,6 +225,18 @@ export default class BlsWalletWrapper {
     return await walletContract.nonce();
   }
 
+  /** Returns the same operation but with a gas estimate. */
+  async opWithGasEstimate(operation: Operation): Promise<Operation> {
+    const bundle = this.sign(operation);
+    operation.gas = this.blockGasLimit;
+    const gatewayAddress = await this.walletContract.trustedBLSGateway();
+    // const gatewaySigner = new VoidSigner(gatewayAddress);
+    operation.gas = await this.walletContract
+      .connect(gatewayAddress)
+      .estimateGas.performOperation(bundle.operations[0]);
+    return operation;
+  }
```

Comment:
If you tweak this like so you can avoid the need to write `gas: 0` and also avoid mutating the operation passed in, both of which are rather confusing.

```suggestion
  async opWithGasEstimate(operation: Omit<Operation, "gas">): Promise<Operation> {
    operation = { ...operation, gas: this.blockGasLimit };
    const bundle = this.sign(operation);
    const gatewayAddress = await this.walletContract.trustedBLSGateway();
    // const gatewaySigner = new VoidSigner(gatewayAddress);
    operation.gas = await this.walletContract
      .connect(gatewayAddress)
      .estimateGas.performOperation(bundle.operations[0]);
    return operation;
  }
```

Reply:
Also, I'm not sure there's much of a use-case for `opWithGasEstimate` without also immediately passing the result to `sign`, so perhaps there should be `signWithGasEstimate` that does both?

Reply:
I did the omit + ... initially, but I think `signWithGasEstimate` will be clearer.